lib/winapi/filemanip.c

```diff
@@ -235,3 +239,80 @@ BOOL MoveFileA (LPCTSTR lpExistingFileName, LPCTSTR lpNewFileName)
         return TRUE;
     }
 }
+
+DWORD GetLogicalDrives (VOID)
+{
+    DWORD result = 0;
+    NTSTATUS status;
+    ANSI_STRING path;
+    HANDLE handle;
+    OBJECT_ATTRIBUTES attributes;
+    struct {
+        OBJECT_DIRECTORY_INFORMATION objDirInfo;
+        CHAR filenameBuf[100];
+    } objDirInfoBuf;
+
+    InitializeObjectAttributes(&attributes, NULL, OBJ_CASE_INSENSITIVE, ObDosDevicesDirectory(), NULL);
+    status = NtOpenDirectoryObject(&handle, &attributes);
+
+    if (!NT_SUCCESS(status)) {
+        SetLastError(RtlNtStatusToDosError(status));
+        return 0;
+    }
+
+    for (ULONG index = 0;;) {
+        POBJECT_DIRECTORY_INFORMATION objDirInfo = (POBJECT_DIRECTORY_INFORMATION)&objDirInfoBuf;
+        status = NtQueryDirectoryObject(handle, objDirInfo, sizeof(objDirInfoBuf), (index == 0), &index, NULL);
+
+        if (status == STATUS_NO_MORE_ENTRIES) {
+            break;
+        } else if (!NT_SUCCESS(status)) {
+            NtClose(handle);
+            SetLastError(RtlNtStatusToDosError(status));
+            return 0;
+        }
+
+        // Ignore entries like "CdRom0:"
```

Comment:
nit: I'd change this comment to `// Only allow entries like "C:" or "E:"` (as this is what the code does)

Reply:
I changed the comment, but kept "CdRom0:" because I think it helps to understand why we're doing this.